url21b471891f_Agetac_Model_tgz-pJ8-org_json_TestJ8/src/org/agetac/common/model/impl/Groupe.java

```diff
@@ -0,0 +1,102 @@
+package org.agetac.common.model.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.agetac.common.exception.InvalidJSONException;
+import org.agetac.common.model.sign.AbstractModel;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class Groupe extends AbstractModel {
+	
+	private Agent chef;
+	private List<Vehicule> vehicules;
+
+	public Groupe() {
+		super(null, null, null);
+		this.chef = null;
+		this.vehicules = new ArrayList<Vehicule>();
+	}
+	
+	public Groupe(String uid, Agent chef, List<Vehicule> vehicules) {
+		super(uid, null, null);
+		
+		this.chef = chef;
+		this.vehicules = vehicules;
+	}
+
+	public Groupe(JSONObject json) throws InvalidJSONException {
+		super(json);
+		try{
+			this.chef = new Agent(json.getJSONObject("chef"));
+	
+			JSONArray jsar = json.getJSONArray("vehicules");
+			vehicules = new ArrayList<Vehicule>();
+			for (int i = 0; i < jsar.length(); i++) {
+				vehicules.add(new Vehicule(jsar.getJSONObject(i)));
+			}
+		}catch(JSONException e){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)